tests/jax/core_system_components/system_component_test.py

```diff
@@ -272,16 +278,37 @@ def system_with_two_components() -> System:
 def test_system_launch_with_build(
     system_with_two_components: System,
 ) -> None:
-    """Test if system can launch having had changed (buildd) the default \
+    """Test if system can launch having had changed (build) the default \
         config.
 
     Args:
         system_with_two_components : mock system
     """
+
     system_with_two_components.build(param_0=2, param_3=False)
     assert system_with_two_components._builder.store.int_plus_str == 3
     assert system_with_two_components._builder.store.float_plus_bool == 1.2
     system_with_two_components.launch()
+    assert system_with_two_components._builder.store.global_config == SimpleNamespace(
+        multi_process=DistributorDefaultConfig.multi_process,
+        name=DistributorDefaultConfig.name,
+        nodes_on_gpu=distributor_default_config.nodes_on_gpu,
+        num_executors=DistributorDefaultConfig.num_executors,
+        param_0=2,
+        param_1=ComponentZeroDefaultConfig.param_1,
+        param_2=ComponentOneDefaultConfig.param_2,
+        param_3=False,
+    )
+
+
+def test_launch_when_not_built(
+    system_with_two_components: System,
+) -> None:
+    """Test that exception is thrown when launch is called before a system\
+        has been built."""
+
+    with pytest.raises(Exception):
```

Comment:
2 things here, in general I think you should expect a more specific exception, but for this case I think we raise a regular `Exception` which I think would be greatly improved if we made a couple custom exceptions:

* `DuplicateConfigError`
* `NotYetBuiltError`
* `AlreadyBuiltError`

Not sure if you want to add it to this PR or make it in a separate PR, but these would likely be pretty common issues for new users and would be nice to have good error messages for them

Reply:
Thanks for the feedback! In the code we are just raising regular Exeptions at the moment, hench the generic checks in the test. I agree with the custom exceptions idea and think it would be an improvement but I think a separate PR for that is probably a better idea. What do you think?

Reply:
Ye they are very descriptive, so maybe it's fine to just leave it. I'm honestly not sure what python convention is, maybe if we keep descriptive error messages then it's fine. Custom exceptions do seem quite _java-like_ so maybe we just leave it